tests/coreTest.php

```diff
@@ -94,4 +95,23 @@ function testSiteSelectionViaCwd() {
     }
     chdir($cwd);
   }
+
+  function testOptionsUri() {
+    // Put a yml file in the drush folder.
+    $drush_directory = sprintf('%s/drush', $this->webroot());
+    $drush_config_file = sprintf('%s/drush.yml', $drush_directory);
+    $test_uri = 'http://test.uri';
+    $options_with_uri = [
+      'options' => [
+        'uri' => $test_uri,
+      ],
+    ];
+    @mkdir($drush_directory);
+    file_put_contents($drush_config_file, Yaml::dump($options_with_uri, PHP_INT_MAX, 2));
+    $this->drush('core-status');
```

Comment:
Don't you need to specify a root and uri for this to work? Thats how we do this all over the test suite. Without that Drush can't find the 'dev' site.

Also, its generally best to send --format=json and then you you dont a regex to get the data you need out. See \Unish\userCase::testBlockUnblock, for an example.

Reply:
That's a good question. I assumed the test was chdir'ing like `testSiteSelectionViaCwd` does. Shouldn't work without some sort of initial site selector. Maybe the chdir from the previous test persists?

Reply:
Well, the output contains the expected URI. I guess if cwd is "project root" it finds it anyway?

I can pass in the root as the other tests of course, to make it more uniform.

Good tip about the json format! The reason I was regexing was because I was looking at the test above this one :)